test/test_ops.py

```diff
@@ -366,47 +370,119 @@ def _to_tensormeta(x):
+                    torch_result,
+                    exact_stride=False,
+                    exact_device=True,
+                    exact_layout=True,
+                    exact_is_coalesced=True,
+                )
+            except AssertionError as e:
+                # Raises the error if the precise dtype comparison wouldn't be
+                # different
+                if dtype is precise_dtype:
+                    raise e
 
-        for sample in op.reference_inputs(device, dtype, requires_grad=False):
-            # Run the test twice: once directly as written...
-            go(sample)
-            # ...and once rerouting all torch.* function calls to torch._refs.*
-            # instead (testing PrimTorch end-to-end)
-            go(sample, lambda: TorchRefsMode.push(strict=True))
-            # TODO: xfail'ing one of these will xfail both, there is no finer
-            # granularity at the moment
+                ex = e
+
+
+            # Goes to next sample if these results are close
+            if not ex:
+                continue
+
+            # If the results are not close, checks that the
+            # reference is more accurate than the torch op
+            def _make_precise(x):
+                if isinstance(x, torch.Tensor) and x.dtype is dtype:
+                    return x.to(precise_dtype)
+                return x
 
+            precise_sample = sample.transform(_make_precise)
+            precise_result = op.torch_opinfo(precise_sample.input, *precise_sample.args, **precise_sample.kwargs)
+
+            def _distance(a, b):
+                # Special-cases boolean comparisons
+                if prims.utils.is_boolean_dtype(a.dtype):
+                    assert b.dtype is torch.bool
+                    return (a ^ b).sum()
+
+                same = (a == b)
+                if prims.utils.is_float_dtype(a.dtype) or prims.utils.is_complex_dtype(a.dtype):
+                    same = torch.logical_or(same, torch.logical_and(torch.isnan(a), torch.isnan(b)))
+
+                actual_error = torch.where(same, 0, torch.abs(a - b)).sum()
+                return actual_error
+
+            ref_distance = 0
+            for a, b in zip(tree_flatten(ref_result)[0], tree_flatten(precise_result)[0]):
+                ref_distance = ref_distance + _distance(a, b)
+
+            torch_distance = 0
+            for a, b in zip(tree_flatten(torch_result)[0], tree_flatten(precise_result)[0]):
+                torch_distance = torch_distance + _distance(a, b)
+
+            msg = f"Reference result was farther ({ref_distance}) from the precise \
+                    computation than the torch result was ({torch_distance})!"
+            self.assertTrue(ref_distance <= torch_distance, msg=msg)
+
+        # Reports numerical accuracy discrepancies
+        if ex is not None:
+            msg = "Test passed because the reference was more accurate than the torch operator."
+            print(msg)
```

Comment:
should this be a warning? Pytest hides stdout of passing tests, so it will be hard to see for people using pytest

Reply:
Sure -- warning it is